controllers/hosting/hostingdeployment_controller_test.go

```diff
@@ -245,39 +241,84 @@ var _ = Describe("Reconciling a HostingDeployment that exists", func() {
 
 		Context("!HasDeletionTimestamp", func() {
 			BeforeEach(func() {
-				mockSageMakerClientBuilder.
-					AddCreateEndpointResponse(sagemaker.CreateEndpointOutput{}).
-					AddDescribeEndpointResponse(CreateDescribeOutputWithOnlyStatus(sagemaker.EndpointStatusCreating))
-
-				shouldHaveDeletionTimestamp = false
-				shouldHaveFinalizer = true
-				shouldHaveEndpointConfig = true
-
 				// Add twice because there are two calls to GetSageMakerEndpointConfigName.
 				sageMakerEndpointConfigNames.PushBack(endpointConfigSageMakerName)
 				sageMakerEndpointConfigNames.PushBack(endpointConfigSageMakerName)
 			})
 
-			It("Creates necessary resources", func() {
-				ExpectNthSubreconcilerCallToKeepUnusedResources(modelReconciler, endpointConfigReconciler, 0)
-			})
+			// Case where the user is restoring the etcd state from an existing endpoint
+			Context("SageMaker endpoint resource exists", func() {
+				BeforeEach(func() {
+					mockSageMakerClientBuilder.
+						AddDescribeEndpointResponse(CreateDescribeOutput(sagemaker.EndpointStatusInService, "outdated-"+endpointConfigSageMakerName)).
+						AddUpdateEndpointResponse(sagemaker.UpdateEndpointOutput{EndpointArn: ToStringPtr("xyz")})
 
-			It("Creates an Endpoint", func() {
+					shouldHaveDeletionTimestamp = false
+					shouldHaveFinalizer = true
+					shouldHaveEndpointConfig = true
+				})
 
-				req := receivedRequests.Front().Next().Value
-				Expect(req).To(BeAssignableToTypeOf((*sagemaker.CreateEndpointInput)(nil)))
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
 
-				createdRequest := req.(*sagemaker.CreateEndpointInput)
-				Expect(*createdRequest.EndpointConfigName).To(Equal(endpointConfigSageMakerName))
-				Expect(*createdRequest.EndpointName).To(Equal(controllercommon.GetGeneratedJobName(deployment.ObjectMeta.GetUID(), deployment.ObjectMeta.GetName(), 63)))
-			})
+				It("Recognises the existing endpoint", func() {
+					ExpectStatusToBe(deployment, string(sagemaker.EndpointStatusInService))
+				})
 
-			It("Requeues after interval", func() {
-				ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				It("Creates new resources", func() {
+					ExpectNthSubreconcilerCallToKeepUnusedResources(modelReconciler, endpointConfigReconciler, 0)
```

Comment:
Could you explain what we are testing in this overall Context please and also specifically line 269.

Reply:
I am not exactly sure, to be honest. This test case, from the comments, ensures that the n-th subreconciler call did not delete the resources. These are the expects that the other test cases use in this scenario.

Reply:
ok for now, but IMO we should review the unit tests for each operator since we understand the operators better. I believe we can clean them up a little bit.